spec/services/courses_query_spec.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe CoursesQuery do
+  let!(:findable_course) do
+    create(:course, :with_full_time_sites)
+  end
+  let!(:another_course) do
+    create(:course, :with_full_time_sites)
+  end
+  let!(:non_findable_course) do
+    create(:course)
+  end
```

Comment:
I chosen consciously to reach the DB instead of double the scope.

I chosen the trade-off of more slow tests over more stability and guarantee progression and behaviour of the course search.

I did via `let!` (let me know if you prefer other way of seeding different courses.

This will grow with the around 15 filters that will be added!

Reply:
I've found that the cleanest way to build these specs is as follows:

```ruby
RSpec.describe CoursesQuery do
  describe ".call" do
    subject(:results) { described_class.call(query) }

    context "when filtering by subject" do
      let!(:maths_subject) { ... }
      let!(:english_subject) { ... }

      let(:query) do
        { subject_ids: [maths_subject.id] }
      end

      it "returns only the courses with matching subject" do
        expect(call).to contain_exactly(maths_subject)
      end
    end
  end
end
```

Group the setup and filters applied next to each other.